src/managed/Microsoft.Extensions.DependencyModel/Microsoft.Extensions.DependencyModel.csproj

```diff
@@ -5,30 +5,52 @@
     <Description>Abstractions for reading `.deps` files.</Description>
     <TargetFrameworks>net451;netstandard1.3;netstandard1.6;netstandard2.0</TargetFrameworks>
     <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netstandard1.3;netstandard1.6;netstandard2.0</TargetFrameworks>
+    <LangVersion>7.3</LangVersion>
   </PropertyGroup>
 
   <Choose>
     <!--
-    Newtonsoft.Json v11.0.1 is the first verison that targets netstandard2.0. Since we added a target for netstandard2.0
-    so users aren't forced to download the 1.x dependencies, it makes sense to use this version of Newtonsoft.Json.
-    However, we still use the previous (v9.0.1) for other TFMs, so existing users don't need to upgrade their Newtonsoft.Json.
+    Since we added a target for netstandard2.0 so users aren't forced to download the 1.x dependencies,
+    or depend on Newtonsoft.Json, use the Microsoft.Bcl.Json.Sources package (which targets netstandard2.0).
+    However, we still use the Newtonsoft.Json (v9.0.1) for other TFMs, so existing users don't need to upgrade their Newtonsoft.Json.
     For example, the SDK targets net4x and is loaded in VS, so it can't upgrade to a new Newtonsoft.Json.
     -->
     <When Condition="'$(TargetFramework)' == 'netstandard2.0'">
+      <!--These properties are necessary to build the source package-->
       <PropertyGroup>
-        <NewtonsoftJsonPackageVersion>11.0.1</NewtonsoftJsonPackageVersion>
+        <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+        <!-- Suppress warnings for unnecessary CLSCompliance attributes -->
+        <NoWarn>3021</NoWarn>
       </PropertyGroup>
+      <ItemGroup>
+        <PackageReference Include="Microsoft.Bcl.Json.Sources" Version="4.6.0-preview.19072.2">
```

Comment:
We should be putting the version numbers in [https://github.com/dotnet/core-setup/blob/master/eng/Versions.props](https://github.com/dotnet/core-setup/blob/master/eng/Versions.props)

Reply:
I followed the format for existing package references within this csproj which included version numbers here. Should they move to Versions.props too?

Reply:
No, the existing package reference version (Newtonsoft) should remain here, since we never want it updated or shared with other libraries.

Putting this version in Versions.props will get it automatically updated through normal dependency flow PRs.

---
In reply to: [251574076](https://github.com/dotnet/core-setup/pull/5009#discussion_r251574076) [](ancestors = 251574076)